Support Live Unit Testing

This is only for developer comfort, not an issue using the library itself.

### Description

Visual Studio Enterprise supports a feature called [Live Unit Testing](https://docs.microsoft.com/en-us/visualstudio/test/live-unit-testing-intro).
Currently this does not work with FluentAssertions.

I've found the fix to get it working, but as it very IDE specific I'm not sure if you allow this fix to be part the project file.

### Complete minimal example reproducing the issue

Press _play_ in the _Live Unit Test_ window.

### Expected behavior:

Function works.
![image](https://user-images.githubusercontent.com/10671831/120101049-5bfa4f80-c144-11eb-975f-049d5e019ddb.png)

### Actual behavior:

Build error as the output path is outside the directory where LUT is running.

Part of the log:
> [12:46:52.741 Verbose] - BuildManager - D:\github\fluentassertions\Tests\FluentAssertions.Specs\Streams\StreamAssertionSpecs.cs(723,33): error CS1061: 'StreamAssertions' does not contain a definition for 'BeWriteOnly' and no accessible extension method 'BeWriteOnly' accepting a first argument of type 'StreamAssertions' could be found (are you missing a using directive or an assembly reference?)
[12:46:52.741 Verbose] - BuildManager - D:\github\fluentassertions\Tests\FluentAssertions.Specs\Streams\StreamAssertionSpecs.cs(576,33): error CS1061: 'StreamAssertions' does not contain a definition for 'NotHaveLength' and no accessible extension method 'NotHaveLength' accepting a first argument of type 'StreamAssertions' could be found (are you missing a using directive or an assembly reference?)
[12:46:52.741 Verbose] - BuildManager - D:\github\fluentassertions\Tests\FluentAssertions.Specs\Streams\StreamAssertionSpecs.cs(625,33): error CS1061: 'StreamAssertions' does not contain a definition for 'BeReadOnly' and no accessible extension method 'BeReadOnly' accepting a first argument of type 'StreamAssertions' could be found (are you missing a using directive or an assembly reference?)
[12:46:53.168 Info] Build completed with failures

### Cause

The `OutputPath` is modified by `FluentAssertions.csproj` to point outside the scope of the LUT building directories.

### Fix

Change `FluentAssertions.csproj`
```xml
<OutputPath>..\..\Artifacts\$(Configuration)\</OutputPath>
```
to
```xml
<OutputPath Condition="'$(BuildingForLiveUnitTesting)' != 'true'">..\..\Artifacts\$(Configuration)\</OutputPath>
```

Another fix might be to set a `BaseOutputPath` in the project file.
See the [LUT FAQ](https://docs.microsoft.com/en-us/visualstudio/test/live-unit-testing-faq) for more details how this works.

I didn't try that one but it does prevent the inclusion of LUT specific variable/condition.
However, the `BuildingForLiveUnitTesting` variable makes the intention of the condition clear.

--- a/Src/FluentAssertions/FluentAssertions.csproj
+++ b/Src/FluentAssertions/FluentAssertions.csproj
@@ -11,7 +11,6 @@
     <CodeAnalysisRuleSet>..\..\Rules.ruleset</CodeAnalysisRuleSet>
     <AllowedOutputExtensionsInPackageBuildOutputFolder>$(AllowedOutputExtensionsInPackageBuildOutputFolder);.pdb</AllowedOutputExtensionsInPackageBuildOutputFolder>
     <EmbedUntrackedSources>true</EmbedUntrackedSources>
-    <OutputPath>..\..\Artifacts\$(Configuration)\</OutputPath>
   </PropertyGroup>
   <PropertyGroup>
     <Authors>Dennis Doomen;Jonas Nyrup</Authors>

--- a/Tests/Approval.Tests/ApiApproval.cs
+++ b/Tests/Approval.Tests/ApiApproval.cs
@@ -34,7 +34,7 @@ namespace Approval.Tests
             var assemblyFile = Path.GetFullPath(
                 Path.Combine(
                     GetSourceDirectory(),
-                    Path.Combine("..", "..", "Artifacts", configurationName, frameworkVersion, "FluentAssertions.dll")));
+                    Path.Combine("..", "..", "Src", "FluentAssertions", "bin", configurationName, frameworkVersion, "FluentAssertions.dll")));
 
             var assembly = Assembly.LoadFile(Path.GetFullPath(assemblyFile));
             var publicApi = assembly.GeneratePublicApi(options: null);

